libs/entitlement/asm-provider/build.gradle

```diff
@@ -14,9 +14,11 @@ dependencies {
   compileOnly project(':libs:core')
   compileOnly project(':libs:logging')
   implementation 'org.ow2.asm:asm:9.7.1'
+  implementation 'org.ow2.asm:asm-util:9.7.1'
+  implementation 'org.ow2.asm:asm-tree:9.7.1'
+  implementation 'org.ow2.asm:asm-analysis:9.7.1'
   testImplementation project(":test:framework")
   testImplementation project(":libs:entitlement:bridge")
-  testImplementation 'org.ow2.asm:asm-util:9.7.1'
 }
 
```

Comment:
nit: You can avoid needing additional license/notice files with the following gradle configuration:

```
tasks.named("dependencyLicenses").configure {
  mapping from: /asm-.*/, to: 'asm'
}
```